src/include/types.h

```diff
@@ -167,6 +173,28 @@ inline ostream& operator<<(ostream& out, const set<A, Comp, Alloc>& iset) {
   return out;
 }
 
+template<class A, class Comp, class Alloc>
+inline ostream& operator<<(ostream& out, const boost::container::flat_set<A, Comp, Alloc>& iset) {
```

Comment:
flat_set and flat_map != set and map

Reply:
I meant ceph::flat_map instead of boost::flat_map, guess I left off too many words!

Reply:
oh!  because they're not aliased in the ceph namespace.  i'm not a real fan of doing that unless there is a reason we'd swap implementations (like we had to with shared_ptr forever ago); it just obscures things for someone reading the code.

Reply:
Oh hmm, I misread the mempool setup and thought you were adding an alias.

I always kind of liked them thanks to the shared_ptr experience. I thought @wjwithagen had taken advantage of that pattern for some porting work as well, but maybe it doesn't matter for boost bits.